tests/utils/component_test_utils.go

```diff
@@ -37,62 +51,84 @@ func getSchemaComponent(components []schema.Component, name string) (*schema.Com
-	setContainerComponentValues(&containerComponent)
-
-	return &containerComponent
+	component := schema.Component{}
+	component.Name = GetRandomUniqueString(8, true)
+	LogInfoMessage(fmt.Sprintf("....... Name: %s", component.Name))
+	component.Container = &schema.ContainerComponent{}
+	devfile.componentAdded(component)
+	return component
 
 }
 
-// createVolumeComponent creates a volume component and set its attribute values
-func createVolumeComponent() *schema.VolumeComponent {
+// createVolumeComponent creates a volume component , ready for attribute setting
+func (devfile *TestDevfile) createVolumeComponent() schema.Component {
 
 	LogInfoMessage("Create a volume component :")
+	component := schema.Component{}
+	component.Name = GetRandomUniqueString(8, true)
+	LogInfoMessage(fmt.Sprintf("....... Name: %s", component.Name))
+	component.Volume = &schema.VolumeComponent{}
+	devfile.componentAdded(component)
+	return component
 
-	volumeComponent := schema.VolumeComponent{}
-	setVolumeComponentValues(&volumeComponent)
+}
 
-	return &volumeComponent
+// AddCommandToContainer adds a command id to a container, creating one if necessary.
+func (devfile *TestDevfile) AddCommandToContainer(commandId string) string {
+
+	LogInfoMessage(fmt.Sprintf("add command %s to a container.", commandId))
+	componentName := ""
+	for _, currentComponent := range devfile.SchemaDevFile.Components {
+		if currentComponent.Container != nil {
+			currentComponent.Container.Command = append(currentComponent.Container.Command, commandId)
+			componentName = currentComponent.Name
+			LogInfoMessage(fmt.Sprintf("add command to existing container : %s", componentName))
+			devfile.componentUpdated(currentComponent)
+			break
+		}
+	}
 
+	if componentName == "" {
+		component := devfile.createContainerComponent()
+		component.Container.Command = append(component.Container.Command, commandId)
+		componentName = component.Name
+		LogInfoMessage(fmt.Sprintf("add command to a new container : %s", componentName))
+		devfile.componentUpdated(component)
+	}
+
+	return componentName
 }
 
 // setContainerComponentValues randomly sets container component attributes to random values
-func setContainerComponentValues(containerComponent *schema.ContainerComponent) {
+func (devfile *TestDevfile) setContainerComponentValues(component *schema.Component) {
+
+	containerComponent := component.Container
 
 	containerComponent.Image = GetRandomUniqueString(8+GetRandomNumber(10), false)
 
 	if GetBinaryDecision() {
 		numCommands := GetRandomNumber(3)
-		containerComponent.Command = make([]string, numCommands)
 		for i := 0; i < numCommands; i++ {
-			containerComponent.Command[i] = GetRandomString(4+GetRandomNumber(10), false)
-			LogInfoMessage(fmt.Sprintf("....... command %d of %d : %s", i, numCommands, containerComponent.Command[i]))
+			commandId := GetRandomString(4+GetRandomNumber(10), true)
+			containerComponent.Command = append(containerComponent.Command, commandId)
```

Comment:
The `command` field under container component is unrelated to `Commands`  defined in devfile. 
`command` and `args` in container are actually the docker container entrypoints(in dockerfile)

This restriction should be removed

Reply:
removed. Main change was when I created an exec command I called a function to add the command to the container - now the function just returns the name of a container without adding the command to it.